<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="yuqimiao" />

<meta name="date" content="2022-11-05" />

<title>TCGA_pancancer_survival</title>

<script src="site_libs/header-attrs-2.1/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DEP_SIMLR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">TCGA_pancancer_survival</h1>
<h4 class="author">yuqimiao</h4>
<h4 class="date">2022-11-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>DEP_SIMLR/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221003code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20221003)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221003code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221003)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyuqimiaoDEPSIMLRtree72c8d0904cd32a6d16b08ce87d585f53bc714eb5targetblank72c8d09a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yuqimiao/DEP_SIMLR/tree/72c8d0904cd32a6d16b08ce87d585f53bc714eb5" target="_blank">72c8d09</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyuqimiaoDEPSIMLRtree72c8d0904cd32a6d16b08ce87d585f53bc714eb5targetblank72c8d09a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/yuqimiao/DEP_SIMLR/tree/72c8d0904cd32a6d16b08ce87d585f53bc714eb5" target="_blank">72c8d09</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/docs/.DS_Store
    Ignored:    analysis/docs/figure/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/Code_collection/.DS_Store
    Ignored:    code/Code_collection/TCGA_survival_analysis/.DS_Store

Untracked files:
    Untracked:  CheckBMIQ-1.pdf
    Untracked:  Type1fit-1.pdf
    Untracked:  Type2fit-1.pdf
    Untracked:  analysis/TCGA_pancancer_survival.rmd
    Untracked:  analysis/simulation_log.rmd
    Untracked:  code/Code_collection/DEP-SIMLR_code_log.md
    Untracked:  code/Code_collection/TCGA_survival_analysis/Package_install.R
    Untracked:  code/Code_collection/TCGA_survival_analysis/SVA_combat.R
    Untracked:  code/Code_collection/TCGA_survival_analysis/methyl_BMIQ_normalization.R
    Untracked:  code/Code_collection/TCGA_survival_pan_cancer_pipeline/
    Untracked:  code/functions/feature_score.R
    Untracked:  code/simulation_scripts/simu_221012_3.R
    Untracked:  code/simulation_scripts/simu_221017_1.R
    Untracked:  code/simulation_scripts/simu_221017_2.R
    Untracked:  code/simulation_scripts/simu_221017_3.R
    Untracked:  data/mutation_dim.rds

Unstaged changes:
    Modified:   analysis/simulation_notebook.rmd
    Modified:   code/functions/visualization_functions.R
    Modified:   code/simulation_scripts/simu_221012_analysis.R

Staged changes:
    Modified:   analysis/simulation_notebook.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This script summarize the results from the TCGA pancancer analysis and pick the cancers with good performance for visualization</p>
</div>
<div id="pancancer-analysis-results-and-store-files-log" class="section level1">
<h1>Pancancer analysis results and store files log</h1>
<p>survival checking tibble</p>
<pre class="r"><code>surv_tib_check</code></pre>
<pre><code># A tibble: 55 × 8
   cancer kernel_type data_type       c method surv_pv sample_size min_clust_size
   &lt;chr&gt;  &lt;chr&gt;       &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;    &lt;dbl&gt;       &lt;dbl&gt;          &lt;dbl&gt;
 1 BLCA   diff_kernel Integration     6 SNF    0.00102         391             51
 2 BLCA   diff_kernel Integration     5 SNF    0.00136         391             61
 3 BLCA   kernel      Integration     5 SNF    0.00213         391             58
 4 BLCA   kernel      Integration     6 SNF    0.00265         391             50
 5 BLCA   kernel      Integration     4 SNF    0.00556         391             69
 6 BLCA   diff_kernel Integration     4 SNF    0.00790         391             70
 7 BLCA   kernel      Integration     3 cimlr  0.0109          391             50
 8 BLCA   kernel      Integration     6 cimlr  0.0133          391             24
 9 BLCA   kernel      Integration     7 SNF    0.0154          391             46
10 BLCA   kernel      Integration     4 cimlr  0.0158          391             51
# … with 45 more rows</code></pre>
<p>Criteria for cancer picking: 1. diff_part_cimlr has better performance 2. smallest cancer have at least 20 subjects ? not right for KIRP 3. integration can find more subtypes</p>
<p>Picking KIRP, KIRC and IUAD</p>
</div>
<div id="visualization" class="section level1">
<h1>Visualization</h1>
<div id="feature-selection-for-visualization" class="section level2">
<h2>Feature selection for visualization</h2>
<p>Using Laplacianc score and robust rank aggregation for feature picking; Specifically, we choose the features consistent with the similarity matrix learned from diff-part-cimlr</p>
</div>
<div id="visualization-for-kirp" class="section level2">
<h2>Visualization for KIRP</h2>
<p>Detailed functions is in ‘survival_res_analysis.R’</p>
<pre class="r"><code>g_kirp = readRDS(&quot;/Volumes/sw2206/yuqi/data_20220308/pancancer_output/alpha_0.8_eigengap/picking_3_cancer/kirp_plots.rds&quot;)</code></pre>
<p>survival tib</p>
<pre class="r"><code>kirp_check</code></pre>
<pre><code># A tibble: 12 × 8
   cancer kernel_type data_type       c method      surv_pv sample_size min_clust_size
   &lt;chr&gt;  &lt;chr&gt;       &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;         &lt;dbl&gt;       &lt;dbl&gt;          &lt;dbl&gt;
 1 KIRP   kernel      expr            2 spectral   5.21e- 2         148             59
 2 KIRP   kernel      me              3 spectral   1.24e- 9         148             34
 3 KIRP   kernel      mu              2 spectral   5.99e- 1         148             74
 4 KIRP   kernel      Integration     4 SNF        1.58e- 1         148             19
 5 KIRP   kernel      Integration     4 cimlr      1.40e- 1         148             23
 6 KIRP   kernel      Integration     4 part_cimlr 1.31e-14         148              9
 7 KIRP   diff_kernel expr            2 spectral   2.63e- 2         148             55
 8 KIRP   diff_kernel me              3 spectral   4.85e-11         148             34
 9 KIRP   diff_kernel mu              2 spectral   5.99e- 1         148             74
10 KIRP   diff_kernel Integration     4 SNF        1.10e- 3         148             26
11 KIRP   diff_kernel Integration     4 cimlr      2.41e- 1         148              9
12 KIRP   diff_kernel Integration     4 part_cimlr 3.38e-14         148              9</code></pre>
<p>surv_plot</p>
<pre class="r"><code>g_kirp$g_sing_surv</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>g_kirp$g_integ_surv</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>heatmap for gene expr</p>
<pre class="r"><code>g_kirp$g_expr</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>heatmap for methylation</p>
<pre class="r"><code>g_kirp$g_me</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="visualization-for-kirc" class="section level2">
<h2>Visualization for KIRC</h2>
<pre class="r"><code>g_kirc = readRDS(&quot;/Volumes/sw2206/yuqi/data_20220308/pancancer_output/alpha_0.8_eigengap/picking_3_cancer/kirc_plots.rds&quot;)</code></pre>
<p>survival tib</p>
<pre class="r"><code>kirc_check</code></pre>
<pre><code># A tibble: 12 × 8
   cancer kernel_type data_type       c method     surv_pv sample_size min_clust_size
   &lt;chr&gt;  &lt;chr&gt;       &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;       &lt;dbl&gt;          &lt;dbl&gt;
 1 KIRC   kernel      expr            2 spectral   9.38e-6         250            116
 2 KIRC   kernel      me              2 spectral   3.88e-4         250             88
 3 KIRC   kernel      mu              4 spectral   5.69e-1         250             42
 4 KIRC   kernel      Integration     3 SNF        9.58e-1         250             29
 5 KIRC   kernel      Integration     3 cimlr      6.57e-1         250             25
 6 KIRC   kernel      Integration     3 part_cimlr 3.47e-1         250             43
 7 KIRC   diff_kernel expr            2 spectral   4.04e-5         250            120
 8 KIRC   diff_kernel me              2 spectral   4.29e-4         250             86
 9 KIRC   diff_kernel mu              4 spectral   6.81e-1         250             41
10 KIRC   diff_kernel Integration     3 SNF        1.14e-1         250             43
11 KIRC   diff_kernel Integration     3 cimlr      6.57e-1         250             25
12 KIRC   diff_kernel Integration     3 part_cimlr 3.16e-2         250             69</code></pre>
<p>surv_plot</p>
<pre class="r"><code>g_kirc$g_sing_surv</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>g_kirc$g_integ_surv</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>heatmap for gene expr</p>
<pre class="r"><code>g_kirc$g_expr</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>heatmap for methylation</p>
<pre class="r"><code>g_kirc$g_me</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="visualization-for-luad" class="section level2">
<h2>Visualization for LUAD</h2>
<pre class="r"><code>g_luad = readRDS(&quot;/Volumes/sw2206/yuqi/data_20220308/pancancer_output/alpha_0.8_eigengap/picking_3_cancer/luad_plots.rds&quot;)</code></pre>
<p>survival tib</p>
<pre class="r"><code>luad_check</code></pre>
<pre><code># A tibble: 12 × 8
   cancer kernel_type data_type       c method     surv_pv sample_size min_clust_size
   &lt;chr&gt;  &lt;chr&gt;       &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;       &lt;dbl&gt;          &lt;dbl&gt;
 1 LUAD   kernel      expr            3 spectral   0.00127         427            101
 2 LUAD   kernel      me              2 spectral   0.276           427            196
 3 LUAD   kernel      mu              3 spectral   0.581           427             52
 4 LUAD   kernel      Integration     4 SNF        0.257           427             55
 5 LUAD   kernel      Integration     4 cimlr      0.0879          427             23
 6 LUAD   kernel      Integration     4 part_cimlr 0.00661         427             23
 7 LUAD   diff_kernel expr            3 spectral   0.00222         427            101
 8 LUAD   diff_kernel me              2 spectral   0.498           427            195
 9 LUAD   diff_kernel mu              2 spectral   0.0437          427             28
10 LUAD   diff_kernel Integration     4 SNF        0.212           427             71
11 LUAD   diff_kernel Integration     4 cimlr      0.0779          427             23
12 LUAD   diff_kernel Integration     4 part_cimlr 0.00588         427             22</code></pre>
<p>surv_plot</p>
<pre class="r"><code>g_luad$g_sing_surv</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>g_luad$g_integ_surv</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>heatmap for gene expr</p>
<pre class="r"><code>g_luad$g_expr</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>heatmap for methylation</p>
<pre class="r"><code>g_luad$g_me</code></pre>
<p><img src="figure/TCGA_pancancer_survival.rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="potential-problems-in-analysis" class="section level1">
<h1>Potential problems in analysis</h1>
<ol style="list-style-type: decimal">
<li>The number of mutation gene per cancer is small.</li>
</ol>
<pre class="r"><code># saveRDS(mutation_dim, file = &quot;./data/mutation_dim.rds&quot;)
mutation_dim = readRDS(&quot;./data/mutation_dim.rds&quot;)

knitr::kable(mutation_dim)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cancer</th>
<th align="right">n_mut</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">BLCA</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">BRCA</td>
<td align="right">57</td>
</tr>
<tr class="odd">
<td align="left">CESC</td>
<td align="right">15</td>
</tr>
<tr class="even">
<td align="left">COAD</td>
<td align="right">188</td>
</tr>
<tr class="odd">
<td align="left">GBM_</td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">HNSC</td>
<td align="right">36</td>
</tr>
<tr class="odd">
<td align="left">KIRC</td>
<td align="right">12</td>
</tr>
<tr class="even">
<td align="left">KIRP</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">LGG_</td>
<td align="right">50</td>
</tr>
<tr class="even">
<td align="left">LIHC</td>
<td align="right">14</td>
</tr>
<tr class="odd">
<td align="left">LUAD</td>
<td align="right">52</td>
</tr>
<tr class="even">
<td align="left">LUSC</td>
<td align="right">15</td>
</tr>
<tr class="odd">
<td align="left">PAAD</td>
<td align="right">81</td>
</tr>
<tr class="even">
<td align="left">PCPG</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">PRAD</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">SARC</td>
<td align="right">15</td>
</tr>
<tr class="odd">
<td align="left">STAD</td>
<td align="right">65</td>
</tr>
<tr class="even">
<td align="left">TGCT</td>
<td align="right">16</td>
</tr>
<tr class="odd">
<td align="left">THCA</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">UCEC</td>
<td align="right">117</td>
</tr>
</tbody>
</table>
<ol start="2" style="list-style-type: decimal">
<li><p>the gene expression data is not normalized; Very large counts in the data, presumably preprocess steps need to be done ;</p></li>
<li><p>Using RRA to rank the features for visualization will give an imbalanced feature ranking, there the clusters sparate further away will dominant those top features;</p></li>
<li><p>Use other source of data</p></li>
</ol>
<p><a href="http://firebrowse.org/" class="uri">http://firebrowse.org/</a></p>
<p><a href="https://github.com/mariodeng/FirebrowseR" class="uri">https://github.com/mariodeng/FirebrowseR</a></p>
<p>microRNA available tractable preprocess</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] survminer_0.4.9 ggpubr_0.4.0    MASS_7.3-51.5   flexsurv_1.1.1 
 [5] survival_3.1-11 forcats_0.5.0   stringr_1.4.0   dplyr_1.0.7    
 [9] purrr_0.3.4     readr_1.3.1     tidyr_1.1.3     tibble_3.1.3   
[13] ggplot2_3.3.5   tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] fs_1.5.0           lubridate_1.7.8    RColorBrewer_1.1-2 httr_1.4.2        
 [5] rprojroot_2.0.2    tools_3.6.2        backports_1.2.1    utf8_1.2.2        
 [9] R6_2.5.0           DBI_1.1.0          colorspace_2.0-2   withr_2.4.2       
[13] gridExtra_2.3      tidyselect_1.1.1   curl_4.3           compiler_3.6.2    
[17] git2r_0.26.1       cli_3.0.1          rvest_0.3.5        xml2_1.3.2        
[21] labeling_0.4.2     scales_1.1.1       survMisc_0.5.5     mvtnorm_1.1-2     
[25] quadprog_1.5-8     digest_0.6.27      foreign_0.8-76     rmarkdown_2.1     
[29] rio_0.5.16         pkgconfig_2.0.3    htmltools_0.5.1.1  highr_0.8         
[33] dbplyr_1.4.2       rlang_0.4.11       readxl_1.3.1       rstudioapi_0.13   
[37] farver_2.1.0       generics_0.1.0     zoo_1.8-9          jsonlite_1.7.2    
[41] zip_2.0.4          car_3.0-7          magrittr_2.0.1     Matrix_1.2-18     
[45] Rcpp_1.0.8.3       munsell_0.5.0      fansi_0.5.0        abind_1.4-5       
[49] lifecycle_1.0.0    stringi_1.5.3      yaml_2.2.1         mstate_0.2.12     
[53] carData_3.0-3      grid_3.6.2         promises_1.2.0.1   crayon_1.4.1      
[57] lattice_0.20-41    cowplot_1.0.0      haven_2.3.1        splines_3.6.2     
[61] hms_1.1.0          knitr_1.32         pillar_1.6.2       ggsignif_0.6.0    
[65] reprex_0.3.0       glue_1.4.2         evaluate_0.14      data.table_1.13.6 
[69] modelr_0.1.6       deSolve_1.28       vctrs_0.3.8        httpuv_1.5.5      
[73] cellranger_1.1.0   gtable_0.3.0       muhaz_1.2.6.1      km.ci_0.5-2       
[77] assertthat_0.2.1   xfun_0.22          openxlsx_4.1.4     xtable_1.8-4      
[81] broom_0.7.6        rstatix_0.7.0      later_1.1.0.1      KMsurv_0.1-5      
[85] workflowr_1.6.2    ellipsis_0.3.2    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
