<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="yuqimiao" />

<meta name="date" content="2022-09-22" />

<title>TCGA_survival_analysis</title>

<script src="site_libs/header-attrs-2.1/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DEP_SIMLR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">TCGA_survival_analysis</h1>
<h4 class="author">yuqimiao</h4>
<h4 class="date">2022-09-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DEP_SIMLR/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221003code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20221003)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221003code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221003)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstronge0d8c2c"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> e0d8c2c </a>
</p>
</div>
<div id="strongRepositoryversionstronge0d8c2c" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version e0d8c2c. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store

Untracked files:
    Untracked:  code/Code_collection/
    Untracked:  code/R/
    Untracked:  code/functions/
    Untracked:  code/simu_220930.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/TCGA_survival_analysis.rmd</code>) and HTML (<code>docs/TCGA_survival_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
53fd9ff
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction-and-raw-data-description" class="section level1">
<h1>Introduction and Raw data description</h1>
<p>In this script, Survival analysis is applied on the kirp and liver cancer data.</p>
<p><strong>KIRP</strong></p>
<ul>
<li>Gene expression:
<ul>
<li>cancer: 16729 290</li>
<li>normal: 16729 30</li>
</ul></li>
<li>Methylation:
<ul>
<li>cancer: 302734 137</li>
<li>normal: 302734 45</li>
</ul></li>
<li>Common sample: 137</li>
</ul>
<p><strong>lihc</strong></p>
<ul>
<li>Gene expression:
<ul>
<li>cancer: 16085 367</li>
<li>normal: 16085 49</li>
</ul></li>
<li>Methylation:
<ul>
<li>cancer: 302747 161</li>
<li>normal: 302747 50</li>
</ul></li>
<li>Common sample: 161</li>
</ul>
</div>
<div id="implementation-detail" class="section level1">
<h1>Implementation detail</h1>
<p>The analysis aims to compare the effect from integration, similarity diffusion and partition-level clustering separately. For each data type in each cancer, Euclidean distance and weighted distance are calculated. Weighted distance is based on definition from Ruan et.al(<span class="math inline">\(d_w(i,j)^2 = \sum_{p=1}^P(x_{ip}-x_{jp})^2w_p, w_p = \frac{-\log_{10}(PV_p)}{\sum_p-\log_{10}(PV_p)}\)</span>. The p value is calculated from the univariate logistic regression with the indicator of the cancer as outcome and each single genetic variant as predictor.</p>
<p>From the distance, we further calculated the kernel and diffused kernel (<span class="math inline">\(K_d = \alpha KP+(1-\alpha)K\)</span>, where P is the knn matrix, <span class="math inline">\(k=\sqrt n\)</span> ) as the similarity matrix reflecting similarity relationship between subjects. and we perform spectral clustering on these similarity measure for each data type in each cancer.</p>
<p>Three methods are implemented for integration: SNF, CIMLR and part-CIMLR. Same kernel lists are fed into the integration methods. We average the kernels in each list and using eigen gap criteria to estimate the total number of clusters in the integrated data. in this way, all 3 integration methods are regarded as having the same number of clusters for the final integraed similarity matrix.</p>
</div>
<div id="result" class="section level1">
<h1>Result</h1>
<pre class="r"><code>surv_pv = readRDS(&quot;./code/Code_collection/TCGA_survival_analysis/surv_pv.rds&quot;)
kirp_surv_pv = surv_pv %&gt;% filter(cancer == &quot;kirp&quot;)
lihc_surv_pv = surv_pv %&gt;% filter(cancer == &quot;lihc&quot;)</code></pre>
<p>Result from lihc is not significant among all analysis, in the “number of cluster tunning” section, we further change the number of clusters of lihc to see if any significant result can be observed</p>
<pre class="r"><code>tib1 = lihc_surv_pv
knitr::kable(tib1, digits = 32)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cancer</th>
<th align="left">dist_type</th>
<th align="left">type</th>
<th align="left">kernel_type</th>
<th align="left">method</th>
<th align="right">est_nclust</th>
<th align="right">surv_pv</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">ge</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">0.3534569</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">ge</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">0.3146513</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">ge</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">0.1229610</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">ge</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">0.2483852</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">me</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">0.3239251</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">me</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">0.4251414</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">me</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">0.3646193</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">me</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">0.2300546</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="right">0.6221907</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="right">0.6189854</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="right">0.3860805</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="right">0.7819822</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="right">0.6741028</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="right">0.2889291</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="right">0.3392436</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="right">0.5674536</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="right">0.2831193</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="right">0.1323802</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="right">0.8809602</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="right">0.2324397</td>
</tr>
</tbody>
</table>
<div id="effect-of-integration" class="section level2">
<h2>Effect of integration</h2>
<pre class="r"><code>tib2 = kirp_surv_pv %&gt;% arrange(dist_type, kernel_type)
knitr::kable(tib2, digits = 32)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cancer</th>
<th align="left">dist_type</th>
<th align="left">type</th>
<th align="left">kernel_type</th>
<th align="left">method</th>
<th align="right">est_nclust</th>
<th align="right">surv_pv</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">ge</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="right">1.248888e-09</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">me</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="right">6.722442e-10</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="right">7.704042e-14</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="right">4.911162e-21</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="right">2.790220e-05</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">ge</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="right">2.618814e-08</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">me</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="right">6.403923e-08</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="right">7.704042e-14</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="right">1.043541e-15</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="right">9.739944e-08</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">ge</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">8.642570e-06</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">me</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="right">2.940859e-07</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="right">2.042054e-05</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="right">3.005323e-14</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="right">8.293289e-06</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">ge</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">5.500565e-06</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">me</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="right">2.791599e-06</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="right">7.505977e-12</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="right">2.365982e-14</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="right">7.741927e-05</td>
</tr>
</tbody>
</table>
</div>
<div id="effect-of-diffusion" class="section level2">
<h2>Effect of diffusion</h2>
<pre class="r"><code>tib3 = kirp_surv_pv %&gt;% arrange(type, dist_type)
knitr::kable(tib3, digits = 32)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cancer</th>
<th align="left">dist_type</th>
<th align="left">type</th>
<th align="left">kernel_type</th>
<th align="left">method</th>
<th align="right">est_nclust</th>
<th align="right">surv_pv</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">ge</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="right">2.618814e-08</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">ge</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="right">1.248888e-09</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">ge</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">5.500565e-06</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">ge</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="right">8.642570e-06</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="right">7.704042e-14</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="right">1.043541e-15</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="right">9.739944e-08</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="right">7.704042e-14</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="right">4.911162e-21</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="right">2.790220e-05</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="right">7.505977e-12</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="right">2.365982e-14</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="right">7.741927e-05</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="right">2.042054e-05</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="right">3.005323e-14</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="right">8.293289e-06</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">me</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="right">6.403923e-08</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">me</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="right">6.722442e-10</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">me</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="right">2.791599e-06</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">me</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="right">2.940859e-07</td>
</tr>
</tbody>
</table>
</div>
<div id="effect-of-partition" class="section level2">
<h2>Effect of partition</h2>
<pre class="r"><code>tib4 = kirp_surv_pv %&gt;%
  filter(method != &quot;single&quot;) %&gt;% 
  arrange(type, dist_type)
knitr::kable(tib4, digits = 32)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cancer</th>
<th align="left">dist_type</th>
<th align="left">type</th>
<th align="left">kernel_type</th>
<th align="left">method</th>
<th align="right">est_nclust</th>
<th align="right">surv_pv</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="right">7.704042e-14</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="right">1.043541e-15</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="right">9.739944e-08</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="right">7.704042e-14</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="right">4.911162e-21</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="right">2.790220e-05</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="right">7.505977e-12</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="right">2.365982e-14</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="right">7.741927e-05</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="right">2.042054e-05</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="right">3.005323e-14</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">dist_w</td>
<td align="left">integration</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="right">8.293289e-06</td>
</tr>
</tbody>
</table>
</div>
<div id="cluster-distribution" class="section level2">
<h2>Cluster distribution</h2>
<pre class="r"><code>cluster_dist = readRDS(&quot;code/Code_collection/TCGA_survival_analysis/cluster_distribution.rds&quot;)

knitr::kable(cluster_dist)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cancer</th>
<th align="left">type</th>
<th align="left">dist_type</th>
<th align="left">kernel_type</th>
<th align="left">method</th>
<th align="right">est_nclust</th>
<th align="left">label</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">1</td>
<td align="right">34</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">2</td>
<td align="right">23</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">3</td>
<td align="right">26</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">4</td>
<td align="right">30</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">5</td>
<td align="right">24</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">1</td>
<td align="right">35</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">2</td>
<td align="right">25</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">3</td>
<td align="right">22</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">4</td>
<td align="right">26</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">5</td>
<td align="left">5</td>
<td align="right">29</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">77</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">ge</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">77</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">32</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">75</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">30</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">33</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">28</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">76</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">36</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">71</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">30</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">37</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">71</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">29</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">80</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">81</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">ge</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">80</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">ge</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">86</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">ge</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">75</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">ge</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">90</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">ge</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">71</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">90</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">71</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">94</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">me</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">67</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">92</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">69</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">96</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">me</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">single</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">65</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">14</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">63</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">72</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">56</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">9</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">39</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">30</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">68</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">59</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">14</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">64</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">109</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">17</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">50</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">70</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">17</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">57</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">64</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">110</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="left">1</td>
<td align="right">50</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="left">2</td>
<td align="right">27</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">3</td>
<td align="left">3</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">62</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">75</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">124</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">13</td>
</tr>
<tr class="even">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">59</td>
</tr>
<tr class="odd">
<td align="left">kirp</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">84</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">77</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">7</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">154</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">99</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">62</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">86</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">75</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">23</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">138</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">101</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">94</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">67</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">131</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">30</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">101</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">74</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">SNF</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">87</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">62</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">part_cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">99</td>
</tr>
<tr class="even">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">1</td>
<td align="right">104</td>
</tr>
<tr class="odd">
<td align="left">lihc</td>
<td align="left">integration</td>
<td align="left">dist_w</td>
<td align="left">diff_kernel</td>
<td align="left">cimlr</td>
<td align="right">2</td>
<td align="left">2</td>
<td align="right">57</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="kirp-exploration" class="section level1">
<h1>KIRP Exploration</h1>
<p>2022-09-27</p>
<p>Read in</p>
<pre class="r"><code>res_tib_integ = readRDS(&quot;/Volumes/sw2206/yuqi/data_20220308/output/benchmark_res/res_tib_integ.rds&quot;)
res_tib_single = readRDS(&quot;/Volumes/sw2206/yuqi/data_20220308/output/benchmark_res/res_tib_single.rds&quot;)
obj_tib = readRDS(&quot;/Volumes/sw2206/yuqi/data_20220308/output/cancer_object/obj_tib.rds&quot;)</code></pre>
<div id="survival-plot" class="section level2">
<h2>Survival plot</h2>
<p>Since the weighted distance may not be accurate enough, all following result using original distance</p>
<p>Integrated</p>
<pre class="r"><code>kirp_surv_tib = res_tib_integ %&gt;%
  filter(dist_type == &quot;dist&quot; &amp; cancer == &quot;kirp&quot;) %&gt;%
  dplyr::select(cancer, method, kernel_type,surv_tib) %&gt;%
  unnest(surv_tib)

ggsurvplot_facet(survfit(Surv(days_to_death, censoring)~cluster,
                         data = kirp_surv_tib),
                 data = kirp_surv_tib,
                 facet.by = c(&quot;kernel_type&quot;, &quot;method&quot;))+
  ggtitle(&quot;Survival curve for kirp using original distance&quot;)</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(&quot;code/Code_collection/TCGA_survival_analysis/kirp_dist_surv.png&quot;,width = 10, height = 6)</code></pre>
<p>This plot shows the survival curv from the integrated similarity facet by kernel type and integrating method</p>
<p>ge</p>
<pre class="r"><code>kirp_surv_tib_ge = res_tib_single %&gt;%
  filter(cancer == &quot;kirp&quot; &amp; dist_type == &quot;dist&quot; &amp; type == &quot;ge&quot;) %&gt;%
  dplyr::select(cancer, kernel_type,surv_tib) %&gt;%
  unnest(surv_tib)
surv_ge = ggsurvplot_facet(survfit(Surv(days_to_death, censoring) ~ cluster, data = kirp_surv_tib_ge),
                 kirp_surv_tib_ge,
                 facet.by = c(&quot;kernel_type&quot;))+
  ggtitle(&quot;Survival curve for kirp using original ge distance&quot;)

surv_ge</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This plot shows the survival curv from the gene expression similarity, facet by kernel type.</p>
<p>me</p>
<pre class="r"><code>kirp_surv_tib_me = res_tib_single %&gt;%
  filter(cancer == &quot;kirp&quot; &amp; dist_type == &quot;dist&quot; &amp; type == &quot;me&quot;) %&gt;%
  dplyr::select(cancer, kernel_type,surv_tib) %&gt;%
  unnest(surv_tib)

surv_me = ggsurvplot_facet(survfit(Surv(days_to_death, censoring)~cluster,
                         data = kirp_surv_tib_me),
                 data = kirp_surv_tib_me,
                 facet.by = c(&quot;kernel_type&quot;))+
  ggtitle(&quot;Survival curve for kirp using original me distance&quot;)
surv_me</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This plot shows the survival curv from the methylation similarity, facet by kernel type.</p>
</div>
<div id="diffusion-effect-explore---kirp-gene-expression-result" class="section level2">
<h2>Diffusion effect explore - KIRP gene expression result</h2>
<pre class="r"><code># pull out the res_list for kirp ge
res_kirp_ge = (obj_tib %&gt;% filter(cancer == &quot;kirp&quot; &amp;type == &quot;ge&quot; &amp; dist_type == &quot;dist&quot; ) %&gt;% pull(object))[[1]]
names(res_kirp_ge)</code></pre>
<pre><code>[1] &quot;distance&quot;               &quot;kernel&quot;                 &quot;diff_kernel&quot;           
[4] &quot;est_nclust_kernel&quot;      &quot;est_nclust_diff_kernel&quot; &quot;cluster_kernel&quot;        
[7] &quot;cluster_diff_kernel&quot;   </code></pre>
<pre class="r"><code># est_n_clust didn&#39;t change after diffusion
res_kirp_ge$est_nclust_kernel</code></pre>
<pre><code>$`Eigen-gap best`
[1] 5

$`Eigen-gap 2nd best`
[1] 2

$`Rotation cost best`
[1] 5

$`Rotation cost 2nd best`
[1] 4</code></pre>
<pre class="r"><code>res_kirp_ge$est_nclust_diff_kernel</code></pre>
<pre><code>$`Eigen-gap best`
[1] 5

$`Eigen-gap 2nd best`
[1] 2

$`Rotation cost best`
[1] 5

$`Rotation cost 2nd best`
[1] 4</code></pre>
<pre class="r"><code># table for the cluster distribution
table(res_kirp_ge$cluster_kernel)</code></pre>
<pre><code>
 1  2  3  4  5 
34 23 26 30 24 </code></pre>
<pre class="r"><code>table(res_kirp_ge$cluster_diff_kernel)</code></pre>
<pre><code>
 1  2  3  4  5 
35 25 22 26 29 </code></pre>
<pre class="r"><code># cluster nmi
compare(res_kirp_ge$cluster_kernel, res_kirp_ge$cluster_diff_kernel,&quot;nmi&quot;)</code></pre>
<pre><code>[1] 0.8771321</code></pre>
<div id="neighborhood-change" class="section level3">
<h3>Neighborhood change</h3>
<p>function for neighborhood check</p>
<pre class="r"><code># input: 2 similarity mat, kernel, diff kernel and the subject id which we want to look at
# output: sorted kernel/diff_kernel vec for the subject, and the plot showing the neighbor difference
neighbor_difference = function(kernel,diff_kernel,subj_id){
  n = nrow(kernel)
  k = floor(sqrt(n))
  par(mfrow = c(2,1))
  plot(kernel[subj_id,], ylim = c(0.7,1))
  plot(diff_kernel[subj_id,], ylim = c(0.7,1))

  sort_kernel_vec = sort(kernel[as.character(subj_id),], index.return = T, decreasing = T)
  sort_diff_kernel_vec = sort(diff_kernel[as.character(subj_id),], index.return = T, decreasing = T)



  par(mfrow = c(1,2))
  plot(x = 1:k, y = sort_kernel_vec$x[1:k],xaxt = &quot;n&quot;, ylab = &quot;kernel&quot;, xlab = &quot;index&quot;, ylim = c(0.7,1), main = paste(&quot;subject&quot;, subj_id, &quot;neighbor change&quot;))
  axis(1, at = 1:k,labels = as.character(colnames(kernel)[sort_kernel_vec$ix[1:k]]),las = 2)
  abline(h=0.85, col = &quot;lightgray&quot;)
  plot(x = 1:k, y = sort_diff_kernel_vec$x[1:k],xaxt = &quot;n&quot;, ylab = &quot;diff_kernel&quot;, xlab = &quot;index&quot;, ylim = c(0.7,1))
  axis(1, at = 1:k,labels = as.character(colnames(kernel)[sort_diff_kernel_vec$ix[1:k]]),las = 2)
  abline(h=0.85, col = &quot;lightgray&quot;)

  return(list(sort_kernel_vec = sort_kernel_vec,
              sort_diff_kernel_vec = sort_diff_kernel_vec))
}</code></pre>
<p>See plot for 2 subjects as example</p>
<pre class="r"><code>tmp = neighbor_difference(kernel = res_kirp_ge$kernel,
                    diff_kernel = res_kirp_ge$diff_kernel,
                    subj_id = &quot;7287&quot;)</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>From this neighborhood comparison, we can see that diffusion makes the neighbor similarity measures less variable, and also change the diagonal similarity measure as the weighted combination of the neighbor similarity;</p>
<p>The neighbor ranks will also change</p>
</div>
<div id="mds-plot-v.s.-survival-plot" class="section level3">
<h3>MDS plot v.s. survival plot</h3>
<p>calc mds plot</p>
<pre class="r"><code>distance = res_kirp_ge$distance
kernel = res_kirp_ge$kernel
diff_kernel = res_kirp_ge$diff_kernel
cluster_kernel = res_kirp_ge$cluster_kernel
cluster_diff_kernel = res_kirp_ge$cluster_diff_kernel

# subset distance measure
colnames(distance) = rownames(distance) = str_sub(colnames(distance), -4,-1)
distance = distance[names(cluster_kernel), names(cluster_kernel)]

distance = dist_kernels(diff_kernel)

mds_kd = as_tibble(cmdscale(distance, 2))%&gt;%
  mutate(kernel = cluster_kernel,
         diff_kernel = cluster_diff_kernel) %&gt;%
  pivot_longer(cols = c(kernel, diff_kernel),
               names_to = &quot;kernel_type&quot;,
               values_to = &quot;cluster&quot;) %&gt;%
  ggplot(aes(x = V1, y = V2, color = factor(cluster))) +
  geom_point()+
  facet_grid(.~kernel_type)+
  labs(title = &quot;MDS plot using diffused kernel distance&quot;,
       subtitle = &quot;labeled by the cluster from kernel and diff_kernel separately&quot;,
       color = &quot;cluster label&quot;)</code></pre>
<pre><code>Warning: The `x` argument of `as_tibble.matrix()` must have unique column names if `.name_repair` is omitted as of tibble 2.0.0.
Using compatibility `.name_repair`.</code></pre>
<pre class="r"><code>mds_kd|surv_ge</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-16-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>plot by kernel distance to show clearer low-dimensional structure</p>
<p>The color of the 2 plots are corresponding; The diffusion kernel clearly separates cluster 3,4, which are the 2 extreme clusters in the population. This indicatesthe fact that diffusion captures more global, more dense and clear cluster structure in the similarity network</p>
<pre class="r"><code>intersect(which(res_kirp_ge$cluster_diff_kernel == 3), c(which(res_kirp_ge$cluster_kernel == 1),which(res_kirp_ge$cluster_kernel == 2)))</code></pre>
<pre><code> [1]   2   8  30  39  42  44  48  52  54  58  59  67  91  92  95 103 112 116 121
[20] 131 132 134</code></pre>
<pre class="r"><code>length(which(res_kirp_ge$cluster_diff_kernel == 3))</code></pre>
<pre><code>[1] 22</code></pre>
<p>Diffusion kernel separate out the most extreme cases as 3/4</p>
</div>
<div id="eigen-value-change-of-gl" class="section level3">
<h3>Eigen value change of GL</h3>
<pre class="r"><code>eig_kernel_GL = eigen(normalized_GL(kernel))
eig_diff_kernel_GL = eigen(normalized_GL(diff_kernel))

eig_kernel_GL$values[(nrow(kernel)-6):nrow(kernel)]</code></pre>
<pre><code>[1] 9.938145e-01 9.927089e-01 9.855136e-01 9.819899e-01 9.791574e-01
[6] 9.739022e-01 1.110223e-16</code></pre>
<pre class="r"><code>eig_diff_kernel_GL$values[(nrow(kernel)-6):nrow(kernel)]</code></pre>
<pre><code>[1]  9.962285e-01  9.952259e-01  9.889565e-01  9.864638e-01  9.827657e-01
[6]  9.771426e-01 -6.661338e-16</code></pre>
<p>smaller eigen value is increasing, focus on larger structure</p>
</div>
</div>
</div>
<div id="subject-6486-observation" class="section level1">
<h1>Subject 6486 observation</h1>
<p>2022-09-28</p>
<div id="common-neighborhood" class="section level2">
<h2>common neighborhood</h2>
<pre class="r"><code>id_cur = &quot;6846&quot;
# obj_tib read in the KIRP exploration
# extract kernel and diff kernel for ge
# pull out the res_list for kirp ge
res_kirp_ge = (obj_tib %&gt;% filter(cancer == &quot;kirp&quot; &amp;type == &quot;ge&quot; &amp; dist_type == &quot;dist&quot; ) %&gt;% pull(object))[[1]]
res_kirp_me = (obj_tib %&gt;% filter(cancer == &quot;kirp&quot; &amp;type == &quot;me&quot; &amp; dist_type == &quot;dist&quot; ) %&gt;% pull(object))[[1]]


kernel_ge = res_kirp_ge$kernel
diff_kernel_ge = res_kirp_ge$diff_kernel

kernel_me = res_kirp_me$kernel
diff_kernel_me = res_kirp_me$diff_kernel</code></pre>
<p>Neighborhood membership in expression</p>
<pre class="r"><code>nb_ge = neighbor_difference(kernel_ge, diff_kernel_ge, subj_id = id_cur)</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-2">
Past versions of unnamed-chunk-20-2.png
</button>
</p>
<div id="fig-unnamed-chunk-20-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Neighborhood membership in methylation</p>
<pre class="r"><code>nb_me = neighbor_difference(kernel_me, diff_kernel_me, subj_id = id_cur)</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-2">
Past versions of unnamed-chunk-21-2.png
</button>
</p>
<div id="fig-unnamed-chunk-21-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ge
nb_ge$sort_kernel_vec$x[1:11]</code></pre>
<pre><code>     6846      A5W9      8515      8310      8630      5891      5156      7268 
1.0000000 0.8613851 0.8610742 0.8604980 0.8524367 0.8512200 0.8481016 0.8477901 
     5876      6133      7288 
0.8470764 0.8454200 0.8445218 </code></pre>
<pre class="r"><code>nb_ge$sort_diff_kernel_vec$x[1:11]</code></pre>
<pre><code>     6846      8515      A5W9      8310      8630      7268      5891      A59R 
0.8922132 0.8565073 0.8560457 0.8522924 0.8502972 0.8485381 0.8476850 0.8446567 
     6133      A55W      A5W7 
0.8445595 0.8411782 0.8410308 </code></pre>
<pre class="r"><code># me
nb_me$sort_kernel_vec$x[1:11]</code></pre>
<pre><code>     6846      8310      8195      7044      7051      A5NJ      7288      7048 
1.0000000 0.8475024 0.8472862 0.8471165 0.8418808 0.8326385 0.8317925 0.8302014 
     A5E6      A55W      7061 
0.8287525 0.8261772 0.8243765 </code></pre>
<pre class="r"><code>nb_me$sort_diff_kernel_vec$x[1:11]</code></pre>
<pre><code>     6846      8195      7044      7051      8310      7288      7048      A5E6 
0.8827232 0.8469597 0.8467426 0.8426171 0.8411741 0.8396802 0.8341499 0.8326273 
     A5NJ      A55W      7061 
0.8317379 0.8259286 0.8219988 </code></pre>
<p>Common neighbors of 6846 in both ge and me</p>
<pre class="r"><code># original neighb of ge/me matrix intersection
intersect(names(nb_ge$sort_kernel_vec$x[1:11]),names(nb_me$sort_kernel_vec$x[1:11]))</code></pre>
<pre><code>[1] &quot;6846&quot; &quot;8310&quot; &quot;7288&quot;</code></pre>
<p>Only subject 8310 and 7288 are both included in the neighbors in gene expression and methylation data</p>
<p>Histogram of days to death</p>
<pre class="r"><code>hist(kirp_surv_tib_ge %&gt;%  filter(kernel_type==&quot;kernel&quot;) %&gt;% pull(days_to_death), decreasing = T, xlab = &quot;days to death&quot;)[1:5]</code></pre>
<pre><code>Warning in plot.window(xlim, ylim, &quot;&quot;, ...): &quot;decreasing&quot; is not a graphical
parameter</code></pre>
<pre><code>Warning in title(main = main, sub = sub, xlab = xlab, ylab = ylab, ...):
&quot;decreasing&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in axis(1, ...): &quot;decreasing&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in axis(2, ...): &quot;decreasing&quot; is not a graphical parameter</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>$breaks
 [1]    0  500 1000 1500 2000 2500 3000 3500 4000 4500 5000 5500 6000

$counts
 [1] 45 50 15  9  7  7  1  2  0  0  0  1

$density
 [1] 6.569343e-04 7.299270e-04 2.189781e-04 1.313869e-04 1.021898e-04
 [6] 1.021898e-04 1.459854e-05 2.919708e-05 0.000000e+00 0.000000e+00
[11] 0.000000e+00 1.459854e-05

$mids
 [1]  250  750 1250 1750 2250 2750 3250 3750 4250 4750 5250 5750

$xname
[1] &quot;kirp_surv_tib_ge %&gt;% filter(kernel_type == \&quot;kernel\&quot;) %&gt;% pull(days_to_death)&quot;</code></pre>
<p>From above plot, we can see that subject 6846 has an extremely long follow up period, this could indicate 6846 to be an outlier of the population;</p>
<div id="mds-plot-v.s.-survival-plot-1" class="section level3">
<h3>MDS plot v.s. survival plot</h3>
<pre class="r"><code># input: 
# output: 
mds_plot = function(distance,
                    kernel,
                    diff_kernel,
                    cluster_kernel,
                    cluster_diff_kernel,
                    target_label){
  
  colnames(distance) = rownames(distance) = str_sub(colnames(distance), -4,-1)
  distance = distance[names(cluster_kernel), names(cluster_kernel)]
  
  distance = dist_kernels(diff_kernel)
  
  
  
  
  mds_kd = as_tibble(cmdscale(distance, 2))%&gt;%
    mutate(kernel = cluster_kernel,
           diff_kernel = cluster_diff_kernel) %&gt;%
    pivot_longer(cols = c(kernel, diff_kernel),
                 names_to = &quot;kernel_type&quot;,
                 values_to = &quot;cluster&quot;) %&gt;%
    arrange(kernel_type) %&gt;% 
    ggplot(aes(x = V1, y = V2, color = factor(cluster), label = rep(target_label,2))) +
    geom_point()+
    geom_text(size = 2)+
    facet_grid(.~kernel_type)+
    labs(title = &quot;MDS plot using diffused kernel distance&quot;,
         subtitle = &quot;labeled by the cluster from kernel and diff_kernel separately&quot;,
         color = &quot;cluster label&quot;)
  
  return(mds_kd)
}</code></pre>
<p>mds+surv, ge/me</p>
<pre class="r"><code>target_label = rownames(kernel)
for(i in 1:length(target_label)){
  target_label[i] = ifelse(target_label[i] %in% names(nb_me$sort_kernel_vec$x[1:11]),
                           target_label[i], 0)
}

mds_me = mds_plot(distance = res_kirp_me$distance,
       kernel = res_kirp_me$kernel,
       diff_kernel = res_kirp_me$diff_kernel,
       cluster_kernel = res_kirp_me$cluster_kernel,
       cluster_diff_kernel = res_kirp_me$cluster_diff_kernel,
       target_label = target_label)

mds_ge = mds_plot(distance = res_kirp_ge$distance,
       kernel = res_kirp_ge$kernel,
       diff_kernel = res_kirp_ge$diff_kernel,
       cluster_kernel = res_kirp_ge$cluster_kernel,
       cluster_diff_kernel = res_kirp_ge$cluster_diff_kernel,
       target_label = target_label)</code></pre>
<pre class="r"><code>(mds_me|mds_ge)/
(surv_me|surv_ge)</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>mds+surv, integrated</p>
<pre class="r"><code>kirp_surv_tib_part = kirp_surv_tib %&gt;%
  filter(method == &quot;part_cimlr&quot;)
kirp_res_tib_integ_part = res_tib_integ %&gt;% 
  filter(cancer ==&quot;kirp&quot; &amp; dist_type == &quot;dist&quot; &amp; method == &quot;part_cimlr&quot;)

mds_part = mds_plot(distance = res_kirp_ge$distance,
       kernel = kirp_res_tib_integ_part$similarity[[1]],
       diff_kernel = kirp_res_tib_integ_part$similarity[[2]],
       cluster_kernel = kirp_res_tib_integ_part$cluster[[1]],
       cluster_diff_kernel =kirp_res_tib_integ_part$cluster[[2]],
       target_label = target_label)

surv_part = ggsurvplot_facet(survfit(Surv(days_to_death, censoring)~cluster,
                         data = kirp_surv_tib_part),
                 data = kirp_surv_tib_part,
                 facet.by = c(&quot;kernel_type&quot;, &quot;method&quot;))+
  ggtitle(&quot;Survival curve for kirp using original distance&quot;)

mds_part/surv_part</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="number-of-cluster-tunning" class="section level1">
<h1>Number of cluster tunning</h1>
<p>Read in, generated from analysis.r</p>
<pre class="r"><code>res_tib_integ_nc = readRDS(&quot;/Volumes/sw2206/yuqi/data_20220308/output/benchmark_res/res_tib_integ_nc.rds&quot;)
surv_pv_integ_nc = readRDS(&quot;/Volumes/sw2206/yuqi/data_20220308/output/benchmark_res/surv_pv_integ_nc.rds&quot;)</code></pre>
<p>&lt;0.05 combination</p>
<pre class="r"><code>surv_pv_integ_nc %&gt;%
  filter(cancer == &quot;lihc&quot;) %&gt;%
  unnest(surv_pv)%&gt;%
  filter(surv_pv&lt;0.05) %&gt;% 
  arrange(surv_pv)</code></pre>
<pre><code># A tibble: 16 × 7
   cancer dist_type type        kernel_type method     n_cluster  surv_pv
   &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;       &lt;chr&gt;       &lt;chr&gt;          &lt;int&gt;    &lt;dbl&gt;
 1 lihc   dist      integration kernel      part_cimlr         6 9.54e-10
 2 lihc   dist_w    integration diff_kernel SNF                3 3.96e- 3
 3 lihc   dist_w    integration kernel      SNF                4 6.48e- 3
 4 lihc   dist_w    integration kernel      part_cimlr         6 6.78e- 3
 5 lihc   dist_w    integration kernel      SNF                3 1.11e- 2
 6 lihc   dist      integration kernel      cimlr              3 1.29e- 2
 7 lihc   dist_w    integration diff_kernel SNF                4 1.73e- 2
 8 lihc   dist      integration kernel      cimlr              4 2.28e- 2
 9 lihc   dist      integration diff_kernel SNF                3 2.91e- 2
10 lihc   dist_w    integration kernel      SNF                5 3.29e- 2
11 lihc   dist      integration diff_kernel cimlr              3 4.06e- 2
12 lihc   dist      integration diff_kernel SNF                4 4.55e- 2
13 lihc   dist      integration diff_kernel SNF                6 4.65e- 2
14 lihc   dist_w    integration kernel      cimlr              4 4.73e- 2
15 lihc   dist      integration kernel      cimlr              6 4.90e- 2
16 lihc   dist      integration kernel      SNF                3 4.95e- 2</code></pre>
<p>lihc survival analysis, nc = 6</p>
<pre class="r"><code>lihc_surv_tib_nc6 = res_tib_integ_nc %&gt;%
  filter(cancer == &quot;lihc&quot; &amp; n_cluster == 6 &amp; dist_type == &quot;dist&quot;) %&gt;%
  dplyr::select(cancer, kernel_type,method, surv_tib) %&gt;%
  unnest(surv_tib) 
surv_6 = ggsurvplot_facet(survfit(Surv(days_to_death, censoring)~cluster,
                                  data = lihc_surv_tib_nc6),
                          data = lihc_surv_tib_nc6,
                          facet.by = c(&quot;kernel_type&quot;, &quot;method&quot;))+
  ggtitle(&quot;Survival curve for lihc using integrated original distance, nc = 6&quot;)

surv_6</code></pre>
<p><img src="figure/TCGA_survival_analysis.rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
18d5b0e
</td>
<td>
yuqimiao
</td>
<td>
2022-10-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Extreme case for surv_6</p>
<pre class="r"><code>part_cim_kernel = lihc_surv_tib_nc6 %&gt;%
  filter(method == &quot;part_cimlr&quot; &amp; kernel_type == &quot;kernel&quot;)
table(part_cim_kernel$cluster)</code></pre>
<pre><code>
 1  2  3  4  5  6 
 1 12  1 33 51 63 </code></pre>
<p>Potentially, when extracting partition information from the single data, only the major variation is captured instead of the small, blur variation.</p>
</div>
<div id="work-log" class="section level1">
<h1>work log</h1>
<p>tidy and add res_tib, surv_tib, cluster_tib into the worklog with header details</p>
</div>
<div id="major-problem-in-the-analysis" class="section level1">
<h1>Major problem in the analysis:</h1>
<ol style="list-style-type: decimal">
<li>num cluster for integrated data is generated from the averaged similarity matrix, which is rural
<ol style="list-style-type: decimal">
<li>In SNF, no need to estimate the num cluster in integ data when fusing</li>
<li>but both cimlr and part-cimlr does</li>
<li>part-cimlr even requires single data num cluster based on the original kernel of each data type, which is easy to go wrong</li>
</ol></li>
<li>Single data analysis should use all the intersection samples instead of all samples</li>
<li>survival data potential problem:
<ol style="list-style-type: decimal">
<li>days to last contact &lt; days to death? <a href="https://www.biostars.org/p/253342/" class="uri">https://www.biostars.org/p/253342/</a></li>
</ol></li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] patchwork_1.0.0 igraph_1.2.6    survminer_0.4.9 ggpubr_0.4.0   
 [5] MASS_7.3-51.5   flexsurv_1.1.1  survival_3.1-11 SNFtool_2.3.0  
 [9] forcats_0.5.0   stringr_1.4.0   dplyr_1.0.7     purrr_0.3.4    
[13] readr_1.3.1     tidyr_1.1.3     tibble_3.1.3    ggplot2_3.3.5  
[17] tidyverse_1.3.0 workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] fs_1.5.0           lubridate_1.7.8    RColorBrewer_1.1-2 httr_1.4.2        
 [5] rprojroot_2.0.2    tools_3.6.2        backports_1.2.1    utf8_1.2.2        
 [9] R6_2.5.0           DBI_1.1.0          colorspace_2.0-2   withr_2.4.2       
[13] gridExtra_2.3      tidyselect_1.1.1   curl_4.3           compiler_3.6.2    
[17] git2r_0.26.1       cli_3.0.1          rvest_0.3.5        xml2_1.3.2        
[21] labeling_0.4.2     scales_1.1.1       survMisc_0.5.5     mvtnorm_1.1-2     
[25] quadprog_1.5-8     digest_0.6.27      foreign_0.8-76     rmarkdown_2.1     
[29] rio_0.5.16         pkgconfig_2.0.3    htmltools_0.5.1.1  prettyGraphs_2.1.6
[33] highr_0.8          dbplyr_1.4.2       rlang_0.4.11       readxl_1.3.1      
[37] rstudioapi_0.13    farver_2.1.0       generics_0.1.0     zoo_1.8-9         
[41] jsonlite_1.7.2     zip_2.0.4          car_3.0-7          magrittr_2.0.1    
[45] heatmap.plus_1.3   Matrix_1.2-18      Rcpp_1.0.8.3       munsell_0.5.0     
[49] fansi_0.5.0        abind_1.4-5        lifecycle_1.0.0    stringi_1.5.3     
[53] whisker_0.4        yaml_2.2.1         carData_3.0-3      mstate_0.2.12     
[57] grid_3.6.2         promises_1.2.0.1   crayon_1.4.1       lattice_0.20-41   
[61] haven_2.3.1        splines_3.6.2      hms_1.1.0          knitr_1.32        
[65] pillar_1.6.2       ggsignif_0.6.0     reprex_0.3.0       glue_1.4.2        
[69] evaluate_0.14      alluvial_0.1-2     data.table_1.13.6  modelr_0.1.6      
[73] deSolve_1.28       vctrs_0.3.8        httpuv_1.5.5       cellranger_1.1.0  
[77] gtable_0.3.0       muhaz_1.2.6.1      km.ci_0.5-2        ExPosition_2.8.23 
[81] assertthat_0.2.1   openxlsx_4.1.4     xfun_0.22          xtable_1.8-4      
[85] broom_0.7.6        rstatix_0.7.0      later_1.1.0.1      KMsurv_0.1-5      
[89] ellipsis_0.3.2    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
